src/back/hlsl/writer.rs

```diff
@@ -131,6 +376,52 @@ impl<W: Write> Writer<W> {
             TypeInner::Scalar { kind, width } => {
                 write!(self.out, "{}", scalar_kind_str(kind, width)?)?;
             }
+            TypeInner::Vector { size, kind, width } => {
+                write!(
+                    self.out,
+                    "{}{}",
+                    scalar_kind_str(kind, width)?,
+                    vector_size_str(size)
+                )?;
+            }
+            TypeInner::Matrix {
+                columns,
+                rows,
+                width,
+            } => {
+                //TODO: int matrix ?
```

Comment:
no need for a TODO - the IR only has float matrices anyway

Reply:
TIL, Thank you!